stdlib/public/core/ArrayShared.swift

```diff
@@ -41,9 +41,14 @@ func _allocateUninitializedArray<Element>(_  builtinCount: Builtin.Word)
   if count > 0 {
     // Doing the actual buffer allocation outside of the array.uninitialized
     // semantics function enables stack propagation of the buffer.
+    let storageType: _ContiguousArrayStorage<Element>.Type
+    #if !$Embedded
```

Comment:
Would it work to sink this down to getContiguousArrayStorageType()?

Reply:
I tried that originally, but the function call that returns a metatype is a problem in embedded Swift. I think we could improve the MandatoryPerfOptimizations to handle that case, but I think I'd like leave that option as a possible follow-up improvement.